File: server/data/results/translations/lite/plain/cpp/collatz_steps.cpp
#include <iostream>
#include <vector>

int collatz_steps(long long n) {
    int steps = 0;
    while (n != 1) {
        if (n % 2 == 0) {
            n = n / 2;
        } else {
            n = 3 * n + 1;
        }
        steps = steps + 1;
    }
    return steps;
}

void main_func() {
    std::vector<long long> starts = {6, 7, 27, 1};
    for (long long start : starts) {
        std::cout << collatz_steps(start) << std::endl;
    }
}

int main() {
    main_func();
    return 0;
}

File: server/data/results/translations/lite/plain/javascript/collatz_steps.js
function collatz_steps(n) {
    let steps = 0;
    while (n !== 1) {
        if (n % 2 === 0) {
            n = Math.floor(n / 2);
        } else {
            n = 3 * n + 1;
        }
        steps = steps + 1;
    }
    return steps;
}

function main() {
    for (const start of [6, 7, 27, 1]) {
        console.log(collatz_steps(start));
    }
}

main();